ui/src/app/services/sign-in.service.ts

```diff
@@ -34,6 +34,20 @@ export class SignInService {
   constructor(private zone: NgZone) {
     this.zone = zone;
 
+    // Set this as early as possible in the application boot-strapping process,
+    // so it's available for Puppeteer to call. If we need this even earlier in
+    // the page, it could go into something like main.ts, but ideally we'd keep
+    // this logic in one place, and keep main.ts minimal.
+    if (environment.allowTestAccessTokenOverride) {
+      window.setTestAccessTokenOverride = (token: string) => {
+        if (token) {
+          window.localStorage.setItem(LOCAL_STORAGE_KEY_TEST_ACCESS_TOKEN, token);
```

Comment:
In theory, I can grab this token and login with cookies, right? Login would be even faster after initial token login.